src/storage/v3/shard_rsm.cpp

```diff
@@ -0,0 +1,443 @@
+
+    if (!vertex_acc) {
+      spdlog::debug(
+          &"Error while trying to delete vertex. Vertex to delete does not exist. Transaction id: "[req.transaction_id
+                                                                                                        .logical_id]);
+      action_successful = false;
+    } else {
+      // TODO(gvolfing)
+      // Since we will not have different kinds of deletion types in one transaction,
+      // we dont have to enter the switch statement on every iteration. Optimize this.
+      switch (req.deletion_type) {
+        case DeleteVerticesRequest::DeletionType::DELETE: {
+          auto result = acc.DeleteVertex(&vertex_acc.value());
+          if (result.HasError() || !(result.GetValue().has_value())) {
+            action_successful = false;
+            spdlog::debug(&"Error while trying to delete vertex. Transaction id: "[req.transaction_id.logical_id]);
+          }
+
+          break;
+        }
+        case DeleteVerticesRequest::DeletionType::DETACH_DELETE: {
+          auto result = acc.DetachDeleteVertex(&vertex_acc.value());
+          if (result.HasError() || !(result.GetValue().has_value())) {
+            action_successful = false;
+            spdlog::debug(
+                &"Error while trying to detach and delete vertex. Transaction id: "[req.transaction_id.logical_id]);
+          }
+
+          break;
+        }
+        default:
+          MG_ASSERT(false, "Non-existent deletion type.");
+      }
+    }
+  }
+
+  DeleteVerticesResponse resp{};
+  resp.success = action_successful;
+
+  if (action_successful) {
+    auto result = acc.Commit(req.transaction_id.logical_id);
+    if (result.HasError()) {
+      resp.success = false;
+      spdlog::debug(&"ConstraintViolation, commiting vertices was unsuccesfull with transaction id: "[req.transaction_id
+                                                                                                          .logical_id]);
+    }
+  }
+
+  return resp;
+}
+
+WriteResponses ShardRsm::ApplyWrite(CreateEdgesRequest &&req) {
+  auto acc = shard_->Access();
+  bool action_successful = true;
+
+  for (const auto &edge : req.edges) {
+    if (!action_successful) {
+      break;
+    }
+
+    auto vertex_acc_from_primary_key = edge.id.src.second;
+    auto vertex_from_acc = acc.FindVertex(ConvertPropertyVector(vertex_acc_from_primary_key), View::OLD);
+
+    auto vertex_acc_to_primary_key = edge.id.dst.second;
+    auto vertex_to_acc = acc.FindVertex(ConvertPropertyVector(vertex_acc_to_primary_key), View::OLD);
+
+    if (!vertex_from_acc || !vertex_to_acc) {
+      action_successful = false;
+      spdlog::debug(
+          &"Error while trying to insert edge, vertex does not exist. Transaction id: "[req.transaction_id.logical_id]);
+      continue;
+    }
+
+    auto edge_type_id = EdgeTypeId::FromUint(edge.type.id);
+    auto edge_acc = acc.CreateEdge(&vertex_from_acc.value(), &vertex_to_acc.value(), edge_type_id);
+
```

Comment:
I forgot to merge #532 and that is necessary here, as it might happen that only one of the vertices are present. I will let you know as soon as I merged the PR, so you can update your PR.